From: unknown author <unknown.author@example.com>
Date: Tue, 1 Nov 2011 18:58:26 +0100
Subject: [PATCH] section comments

http://sources.redhat.com/ml/binutils/2004-04/msg00665.html

fixes building on some architectures (like m68k/arm/cris/etc...) because
it does the right thing

This patch was taken from gentoo.
---
 include/libc-symbols.h |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/libc-symbols.h b/include/libc-symbols.h
index 67e1ca2..8cefa67 100644
--- a/include/libc-symbols.h
+++ b/include/libc-symbols.h
@@ -240,12 +240,12 @@
 #  define __make_section_unallocated(section_string)
 # endif
 
-/* Tacking on "\n\t#" to the section name makes gcc put it's bogus
+/* Tacking on "\n#APP\n\t#" to the section name makes gcc put it's bogus
    section attributes on what looks like a comment to the assembler.  */
 # ifdef HAVE_SECTION_QUOTES
-#  define __sec_comment "\"\n\t#\""
+#  define __sec_comment "\"\n#APP\n\t#\""
 # else
-#  define __sec_comment "\n\t#"
+#  define __sec_comment "\n#APP\n\t#"
 # endif
 # define link_warning(symbol, msg) \
   __make_section_unallocated (".gnu.warning." #symbol) \
